### PR TITLE
[Merged by Bors] - feat: cache accepts module names as well as file names

### PR DESCRIPTION
**Pull request by kim-em**
My most common use case for `lake exe cache get ARG` is with `ARG` an import in a file I have open. This allows you to just copy and paste the import, without having to change `.` to `/` and add `.lean`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Cache/Main.lean [hunk 1]
```diff
@@ -43,12 +43,16 @@ Which will download the cache for:
...
 /-- Note that this normalizes the path strings, which is needed when running from a unix shell
 (which uses `/` in paths) on windows (which uses `\` in paths) as otherwise our filename keys won't
 match. -/
 def toPaths (args : List String) : List FilePath :=
-  args.map (FilePath.mk · |>.normalize)
+  args.map fun arg =>
+    if arg.endsWith ".lean" then
+      FilePath.mk arg |>.normalize
+    else
+      mkFilePath (arg.splitOn ".") |>.withExtension "lean"
```
Comment (jcommelin) [inline Cache/Main.lean:55]: I guess there is a small risk that theoretically this doesn't work with imports where there is a `.` that is escaped by french quotes?
Should that be mentioned in a comment?

Can we alternatively invoke some existing Lean function that turns module names into filepaths?

Comment (semorrison) [inline Cache/Main.lean:55]:
```suggestion
      -- Note that this is not robust to names with French-quoted `.`s
      mkFilePath (arg.splitOn ".") |>.withExtension "lean"
```

Comment (semorrison) [reply Cache/Main.lean]: There's not a single step, but I've replaced it with 
```
mkFilePath (arg.toName.components.map Name.toString) |>.withExtension "lean"
```
which handles French quotes.

Comment (jcommelin) [reply Cache/Main.lean]: Thanks, LGTM

**Review by jcommelin — APPROVED** [review]

File: Cache/Main.lean [hunk 2]
```diff
@@ -44,12 +44,16 @@ Which will download the cache for:
...
 (which uses `/` in paths) on windows (which uses `\` in paths) as otherwise our filename keys won't
 match. -/
-def toPaths (args : List String) : List FilePath :=
-  args.map (FilePath.mk · |>.normalize)
+def toPaths (args : List String) : IO (List FilePath) :=
+  args.mapM fun arg =>
+    if arg.endsWith ".lean" then
+      return FilePath.mk arg |>.normalize
+    else do
+      return mkFilePath (arg.toName.components.map Name.toString) |>.withExtension "lean"
```
Comment (digama0) [inline Cache/Main.lean:56]: why is this in IO?

Comment (semorrison) [reply Cache/Main.lean:56]: Oops. It had been when I was using some Lean internal function that needed to be in IO, then never cleaned up.

**Comment by alexjbest** [comment]
maintainer merge

**Pull request closed without merging**